### ISSUE TITLE
Is it possible to have crossorigin attribute when loading ArcGIS modules?

### ISSUE THREAD

**Issue by StreetStrider** [Esri/esri-loader#281]
Hello, I'm not sure is this related to esri-loader especially, but this is related to ArcGIS JS in general. If this is not the proper place for a ticket, please, give me right direction.
### Expected behavior
We use esri-loader + arcgis modules in JS and we want to receive detailed information on errors when they originate from arcgis. In order to have that we need proper CORS info set up and `crossorigin=anonymous` attribute for arcgis scripts.
Using devtools Network tab I can tell that `js.arcgis.com` domain already sending proper `access-control-allow-` headers and we've already set `crossorigin=anonymous` for our arcgis entry point and esri-loader. But the modules which is loaded asynchronously by loader itself does not feature that attribute. That means for us that if error is originating from one of arcgis modules (which potentially might be the case) we would lost stacktrace information and other error details.
Is it possible for ArcGIS system to set `crossorigin=anynomous` for loaded modules?

We use `<script src="https://unpkg.com/esri-loader" crossorigin="anonymous"></script>` and `<script src="https://js.arcgis.com/4.14" crossorigin="anonymous"></script>`.

### Actual behavior
We receive some cryptic errors in our telemetry, they look like `ScriptError` or `RangeError: maximum call stack size exceeded` without actual callstacks. Since our own errors got detailed info, our best bet is that non-detailed errors are caused by code from third party domains (or at least some of them). This is in agreement with [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-crossorigin).

### Steps to reproduce the behavior
There's no precise reproduction, since I'm expecting some errors which does not available on demand. But scenario in general would be: include esri-loader and arcgis and load some gis component. The loaded modules would not feature `crossorigin=anonymous` attribute, because they loaded as a result of dependency resolving by arcgis itself.

**Comment by andygup** [comment]
Hi @StreetStrider the short answer is no, there aren't any guaranteed client-side workarounds for this issue. We don't have any plans to change the behavior in the AMD CDN build. For troubleshooting, one suggestion is to tell Chrome to stop at all errors. 

The only definitive fix would be to upgrade/migrate to @arcgis/core ES modules at some point.

**Comment by andygup** [comment]
> But it seemes that this is not the only place that creates script elements, there's another place inside modules that comes from arcgis itself.

Right, you could potentially monkey patch `require.injectUrl` but it comes with the same challenges and caveats: https://github.com/dojo/dojo/blob/master/dojo.js#L1746

**Comment by StreetStrider** [comment]
I've tried to use `@arcgis/core@4.18` and got this error:
```
WARNING in ../../node_modules/@arcgis/core/core/workers/workers.js 5:863-872
Critical dependency: the request of a dependency is an expression
```
As it seemes through the code, `arcgis/core` uses dynamic imports with variable. Is it possible to consume `arcgis/core` by webpack alone or do I need some special plugin to assist?

**Comment by odoe** [comment]
That webpack warning can be ignored, it has no impact on build tooling. You can turn them off in webpack here.
https://webpack.js.org/configuration/stats/
There is no special plugin needed for `@arcgis/core`, but this is outside the scope of this repo.

**Comment by StreetStrider** [comment]
@odoe thank you for answer.

> but this is outside the scope of this repo

Understood. I expected it to be so. Where is the better place to address `arcgis/core` related questions?

**Comment by odoe** [comment]
The resources repo might be a better place.
https://github.com/Esri/jsapi-resources/

**Comment by andygup** [comment]
Closing as won't fix.
